Framework/Parallel/src/IO/EventsListsShmemStorage.cpp

```diff
@@ -0,0 +1,61 @@
+#include "MantidParallel/IO/EventsListsShmemStorage.h"
+#include "MantidTypes/Event/TofEvent.h"
+#include <memory>
+
+namespace Mantid {
+namespace Parallel {
+namespace IO {
+
+/// Constructor
+EventsListsShmemStorage::EventsListsShmemStorage(const std::string &segmentName,
+                                                 const std::string &elName,
+                                                 size_t size, size_t chunksCnt,
+                                                 size_t pixelsCount,
+                                                 bool destroy)
+    : EventsListsShmemManager(segmentName, elName, 0), destroyShared(destroy) {
+  try {
+    m_segment = std::make_shared<ip::managed_shared_memory>(
+        ip::create_only, m_segmentName.c_str(), size);
+    m_allocatorInstance =
+        std::make_shared<VoidAllocator>(m_segment->get_segment_manager());
+    m_chunks = m_segment->construct<Chunks>(m_chunksName.c_str())(
+        chunksCnt, EventLists(pixelsCount, EventList(alloc()), alloc()),
+        alloc());
+  } catch (std::exception const &ex) {
+    std::cout << ex.what() << "\n";
+    std::rethrow_if_nested(ex);
+  }
+}
+
+/// Destructor
+EventsListsShmemStorage::~EventsListsShmemStorage() {
+  if (destroyShared)
+    ip::shared_memory_object::remove(m_segmentName.c_str());
```

Comment:
Isn't this allocated in the base class? Shouldn't the base class do the deallocation then? We is this controlled by a flag `destroy`?  The flag has no influence on the *allocation* in the constructor, why does it affect *deallocation*?

Reply:
This is not allocated in the base class. The flag is used for the case then  shared memory segment created in child process and has to me destroyed in parent process.